src/utils.ts

```diff
@@ -64,14 +64,14 @@ export function max(a: BigNumberish, b: BigNumberish): BN {
 export const fixedPointAdjustment = toBNWei("1");
 
 /**
- * Convert an amount of eth into a token given price and token decimals.
+ * Convert an amount of native gas token into a token given price and token decimals.
  *
- * @param {BigNumberish} fromAmount - Amount of eth to convert.
- * @param {string | number} [ price=1 ] - The price as eth per token, ie how much eth can 1 token buy.
+ * @param {BigNumberish} fromAmount - Amount of native gas token to convert.
+ * @param {string | number} [ price=1 ] - The price as native gas token per token, ie how much native gas token can 1 token buy.
  * @param {} [ toDecimals=18 ] - Number of decimals for the token currency.
  * @returns {string} The number of tokens denominated in token decimals in the smallest unit (wei).
  */
-export function ethToToken(fromAmount: BigNumberish, price: string | number = 1, toDecimals = 18): string {
+export function nativeToToken(fromAmount: BigNumberish, price: string | number = 1, toDecimals = 18): string {
```

Comment:
Does this function assume that the gas token has 18 decimals?

Reply:
it does, yea. will need to change this to not assume native gas decimals